nav2_controller/include/nav2_controller/plugins/position_goal_checker.hpp

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2025 Prabhav Saxena
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_CONTROLLER__PLUGINS__POSITION_GOAL_CHECKER_HPP_
+#define NAV2_CONTROLLER__PLUGINS__POSITION_GOAL_CHECKER_HPP_
+
+#include <string>
+#include <memory>
+#include <vector>
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp_lifecycle/lifecycle_node.hpp"
+#include "nav2_core/goal_checker.hpp"
+#include "nav2_costmap_2d/costmap_2d_ros.hpp"
+
+namespace nav2_controller
+{
+
+/**
+ * @class PositionGoalChecker
+ * @brief Goal Checker plugin that only checks XY position, ignoring orientation
+ */
+class PositionGoalChecker : public nav2_core::GoalChecker
```

Comment:
Needs to be added in docs to this list with its configuration page: https://docs.nav2.org/configuration/packages/configuring-controller-server.html#provided-plugins. Here as well: https://docs.nav2.org/plugins/index.html.